lib/SIL/SILType.cpp

```diff
@@ -309,15 +309,28 @@ bool SILType::canRefCast(SILType operTy, SILType resultTy, SILModule &M) {
 }
 
 SILType SILType::getFieldType(VarDecl *field, SILModule &M) const {
-  assert(field->getDeclContext() == getNominalOrBoundGenericNominal());
+  auto baseTy = getSwiftRValueType();
```

Comment:
Why doesn't getTypeOfMember() handle this correctly?

Reply:
Good point, it ought to. It should be sufficient to just remove this assertion.